vega/base.py

```diff
@@ -51,6 +53,8 @@ def _ipython_display_(self):
             metadata={'jupyter-vega': '#{0}'.format(id)}
         )
 
-    def display(self):
+    def display(self, renderer=None):
         """Render the visualization."""
+        if renderer:
+            self.renderer = renderer
```

Comment:
This should only get set in the `__init__`, otherwise this is a side effect on the class when displayed.